metagraph/src/annotation/representation/column_compressed/annotate_column_compressed.cpp

```diff
@@ -234,27 +234,39 @@ void ColumnCompressed<Label>::serialize_counts(const std::string &filename) cons
 
     for (size_t j = 0; j < relation_counts_.size(); ++j) {
         if (!relation_counts_[j].size()) {
-            sdsl::int_vector<>(bitmatrix_[j]->num_set_bits(), 0, count_width_).serialize(out);
+            sdsl::int_vector<>(bitmatrix_[j]->num_set_bits(), 0, 1).serialize(out);
 
         } else {
             assert(relation_counts_[j].size() == bitmatrix_[j]->num_set_bits()
                 && "Binary relation matrix must not be changed while adding relation counts");
 
+            // pack counts
+            uint64_t max_count = 0;
             for (uint64_t v : relation_counts_[j]) {
                 num_counts++;
                 sum_counts += v;
+                max_count = std::max(max_count, v);
             }
 
-            relation_counts_[j].serialize(out);
+            const uint8_t packed_width = sdsl::bits::hi(max_count) + 1;
+            if (packed_width == relation_counts_[j].width()) {
+                relation_counts_[j].serialize(out);
+            } else {
+                // TODO: use int_vector_buffer<>
+                sdsl::int_vector<> packed_counts(relation_counts_[j].size(), 0, packed_width);
+                std::copy(relation_counts_[j].begin(), relation_counts_[j].end(),
+                          packed_counts.begin());
+                packed_counts.serialize(out);
```

Comment:
if you want to save a bit more space doing this, check out sdsl's implementation of `bit_compress`, they do this in-place:

https://github.com/simongog/sdsl-lite/blob/master/include/sdsl/util.hpp#L403

Reply:
Yeah, I also wanted to do that first, but this method is `const`, so I can't change the width of the existing vector. That's why you see this workaround.